packages/components/src/FilterBar/FilterBar.spec.tsx

```diff
@@ -91,66 +111,182 @@ const FilterBarWrapper = (
+      await waitFor(() => {
+        expect(filterButton).not.toBeInTheDocument()
+      })
+
+      const addFiltersButton = getByRole("button", { name: "Add Filters" })
+      await user.click(addFiltersButton)
+
+      const list = getByRole("list")
+      const menuOption = within(list).getByRole("button", { name: "Topping" })
+      await waitFor(() => {
+        expect(menuOption).toBeVisible()
+      })
+
+      await user.click(menuOption)
+      await waitFor(() => {
+        expect(list).not.toBeInTheDocument()
+      })
+      expect(getByRole("button", { name: "Topping" })).toBeVisible()
+    })
+
+    it("adds new filters in the provided order", async () => {
+      const { getByRole, getAllByTestId } = render(
+        <FilterBarWrapper<ValuesSimple>
+          filters={simpleFilters.map(filter => ({
+            ...filter,
+            isRemovable: true,
+          }))}
+        />
+      )
+
+      const addFiltersButton = getByRole("button", { name: "Add Filters" })
+      await user.click(addFiltersButton)
+
+      const menuOptionIceLevel = getByRole("button", { name: "Ice Level" })
+      await user.click(menuOptionIceLevel)
+      await user.click(addFiltersButton)
+
+      const menuOptionFlavour = getByRole("button", { name: "Flavour" })
+      await user.click(menuOptionFlavour)
+      await user.click(addFiltersButton)
+
+      const menuOptionSugarLevel = getByRole("button", { name: "Sugar Level" })
+      await user.click(menuOptionSugarLevel)
+
+      const filters = getAllByTestId(TEST_ID__FILTER)
+      expect(filters.length).toBe(3)
+      expect(filters[0]).toHaveTextContent("Ice Level")
+      expect(filters[1]).toHaveTextContent("Flavour")
+      expect(filters[2]).toHaveTextContent("Sugar Level")
+    })
+  })
+
+  describe("External events", () => {
+    it("allows updating the values via an external event", async () => {
+      const Wrapper = (): JSX.Element => {
+        type ExternalEventValues = {
+          flavour: string
+        }
+
+        const [values, setValues] = useState<Partial<ExternalEventValues>>({})
+
+        const filters = [
+          {
+            id: "flavour",
+            name: "Flavour",
+            Component: (
+              <FilterBar.Select
+                items={[
+                  { value: "honey-milk-tea", label: "Honey Milk Tea" },
+                  { value: "lychee-green-tea", label: "Lychee Green Tea" },
+                ]}
+              />
+            ),
+          },
+        ] satisfies Filters<ExternalEventValues>
+
+        return (
+          <div>
+            <FilterBar
+              filters={filters}
+              values={values}
+              onValuesChange={setValues}
+            />
+            <button
+              type="button"
+              onClick={() => setValues({ flavour: "honey-milk-tea" })}
+            >
+              Update Flavour to honey-milk-tea
+            </button>
```

Comment:
Instead of an actual button here, can't you just trigger `setValues` programmatically?

Reply:
Yes, but this is replicating a user's behaviour. Ensuring that an external event works.